synapse/util/caches/response_cache.py

```diff
@@ -126,7 +126,7 @@ def _set(
         key = context.cache_key
         self.pending_result_cache[key] = result
 
-        def on_complete(r):
+        def on_complete(r: Any) -> Any:
```

Comment:
nit: `def on_complete(r: T) -> T` would be more precise

Reply:
Agreed. I think we might be able to go further though: I think `r: RV`. Reasoning:

- `wrap`'s callback returns an `Awaitable[RV]`

https://github.com/matrix-org/synapse/blob/894be5fb9a9b6fa4d5ef3d4daf1926975d26b303/synapse/util/caches/response_cache.py#L148-L155

- That callback is `run_in_background` to build a `Deferred`, which is passed to `_set`

https://github.com/matrix-org/synapse/blob/894be5fb9a9b6fa4d5ef3d4daf1926975d26b303/synapse/util/caches/response_cache.py#L200-L201

- That deferred's result is what gets passed to on_complete